### ISSUE TITLE
IDE0072: False positive when using switch expression on integer

### ISSUE THREAD

**Issue by omarjuul** [dotnet/roslyn#66433]
**Version Used**: 
.NET SDK 7.0.102

**Steps to Reproduce**:
`dotnet build` the following:

Program.cs
```C#
static string TestSwitch(int foo)
    => foo switch
    {
        0 => "zero",
        < 0 => "negative",
        > 0 => "positive",
    };
```

ConsoleApp0.csproj
```XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>

</Project>
```

.editorconfig
```EditorConfig
[*.cs]
dotnet_diagnostic.IDE0072.severity = warning
```

**Diagnostic Id**: IDE0072

**Expected Behavior**:

IDE0072 is not raised: all cases are handled

**Actual Behavior**:
IDE0072 is raised. The suggested fix is to add a default arm. When I do that, I get a message that (correctly) informs me that `The pattern has already been handled by a previous arm of the 'switch' expression`

**Comment by Youssef1313** [comment]
Similar issue: https://github.com/dotnet/roslyn/issues/60784

**Comment by AdamSpeight2008** [comment]
The compiler doesn't have enough knowledge to determinate that on only a type level. 
A types behavioral traits or effects are type specific and are not expressed in a way, that can effect compilation or validity of a program at compile-time.

Eg
For types that can consider to be "unsigned values"
Two fundemental conditions have to be satifised to met "all cases handled". 

  *  The value Zero has be covered.
  *  All other values greater than Zero has to be covered.
For "signed values" and addition condition has to be met
  *  All other values lesser than Zero has to covered.

This hard to tell just looking at the type and the operators it provides. As the operators `<` `=` `>` are not enough to assert that those conditions are met. A type is allowed to override the meaning, of them. 
The compiler would be required to have specific knowledge of each type, and its intended meaning of each operator. Or that the type providers an interface or attribute, testifying to meaning.

**Comment by omarjuul** [comment]
Somehow the origin of the message that pops up when I apply the suggested fix *does* have this (type-specific) information.

This message seems to be originating from a different analyzer altogether. Is it just a regular build Error? `dotnet build` indicates
 ```
error CS8510: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.
```

**Comment by CyrusNajmabadi** [comment]
Because we didn't have fix all for refactorings at the time. Now that we do, this can be converted to that.
